### ISSUE TITLE
Sonarr doesn't recover after qBittorrent error is resolved

### ISSUE THREAD

**Issue by agneevX** [Sonarr/Sonarr#4674]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a release is sent to qBittorrent that errors for whatever reason (low disk space, network), Sonarr blacklists the release, prevents it from showing in queue, while in History it marks it as failed with a message, `qBittorrent is reporting an error`, which isn't accurate. 

If I resolve the issue in qBittorrent (and it has successfully downloaded), Sonarr continues to ignore the release and keeps it blacklisted while showing the same message in history. 

Restarting Sonarr doesn't do anything either (in Radarr, imports are successful after a restart). 

The only way is to manually import it or delete and re-add it entirely. 

### Expected Behavior

1. Sonarr should not blacklist the release or remove it from queue, despite it being in a state of "error". 
2. Once the error is resolved and the release has downloaded, it should import it automatically, and the entries in History should no longer be marked as failures. 

### Steps To Reproduce

1. Download a release that produces an error in qBittorrent (low disk space)
2. Notice how Sonarr does what I described above. 

### Environment

```markdown
- OS: Ubuntu 20.04
- Sonarr: 3.0.6.1343
- Docker Install: Yes
- Using Reverse Proxy: Yes
- Browser:
```

### What branch are you running?

Develop

### Anything else?

-
